mongo/integration/retryable_writes_prose_test.go

```diff
@@ -284,4 +284,96 @@ func TestRetryableWritesProse(t *testing.T) {
 			// Assert that the "ShutdownInProgress" error is returned.
 			require.True(mt, err.(mongo.WriteException).HasErrorCode(int(shutdownInProgressErrorCode)))
 		})
+
+	mtOpts = mtest.NewOptions().Topologies(mtest.Sharded).MinServerVersion("4.2")
+	mt.RunOpts("retrying in sharded cluster", mtOpts, func(mt *mtest.T) {
+		tests := []struct {
+			name string
+
+			// Note that setting this value greater than 2 will result in false
+			// negatives. The current specification does not account for CSOT, which
+			// might allow for an "inifinite" number of retries over a period of time.
+			// Because of this, we only track the "previous server".
```

Comment:
Is there a task for updating the retryable reads/writes "deprioritized mongos" behavior to account for multiple retries (i.e. CSOT)? The vast majority of sharded clusters have >2 mongos nodes, so that seems like a questionably useful feature for drivers that support CSOT.

Reply:
For now, there is no task to do this. Here are a couple of reasons from my discussions with @comandeo:

> - We do not want this new mechanism to replace SDAM/interfere with SDAM too much.
>- We believe that mongos may recover from the error fast enough, and there is no reason to exclude ones that failed earlier
>- It is rather a rare occasion that multiple mongoses fail with retryable errors. This looks like a network issue, and this is handled by SDAM